core/templates/pages/donate-page/donate-page.component.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the donate page.
+ */
+
+import { Component, OnInit } from '@angular/core';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
+import { SiteAnalyticsService } from 'services/site-analytics.service.ts';
+import { WindowDimensionsService } from
+  'services/contextual/window-dimensions.service.ts';
+
+require('base-components/base-content.directive.ts');
+
+@Component({
+  selector: 'donate-page',
+  template: require('./donate-page.component.html'),
+  styleUrls: []
+})
+export class DonatePageComponent implements OnInit {
+  windowIsNarrow: boolean = false;
+  donateImgUrl: string = '';
+  constructor(
+    private siteAnalyticsService: SiteAnalyticsService,
+    private urlInterpolationService: UrlInterpolationService,
+    private windowDimensionService: WindowDimensionsService
+  ) {}
+  ngOnInit() {
+    this.windowIsNarrow = this.windowDimensionService.isWindowNarrow();
+    this.donateImgUrl = this.urlInterpolationService.getStaticImageUrl(
+      '/general/opp_donate_text.svg');
+  }
+
+  onDonateThroughAmazon() {
+    this.siteAnalyticsService.registerGoToDonationSiteEvent('Amazon');
+    window.location.href = 'https://smile.amazon.com/ch/81-1740068';
```

Comment:
@bansalnitish, there was a timeout for 150ms which I had ignored, maybe we should add it, not really sure what the timeout was doing in the first place.

Reply:
I can answer this one. We had some issues with the call to siteAnalyticsService not firing before the page was redirected, hence we added a little timeout to make sure the call actually happened. I'm not sure if this is necessary in Angular; it might be worth testing.

Reply:
@seanlip couldn't we have gone for this instead?
https://developers.google.com/analytics/devguides/collection/analyticsjs/sending-hits#hitcallback

Reply:
It looks good. It probably wasn't around at the time but I'm not opposed to refactoring accordingly. You'll probably want to that in a second PR though.

Reply:
Do we really want what @srijanreddy98 suggested since it's working fine for me locally?

Reply:
@bansalnitish What is working fine exactly (what did you check)?

Reply:
@seanlip, please correct me if I am wrong. In order to check this, we will have to first change `constants.CAN_SEND_ANALYTICS_EVENTS` to `true` and either check the GA dashboard or use the hitcallback to see if we get the request was sent. We could also check the networks tab.

Reply:
I've added the timeout as per our offline discussion.